File: WP7RxSamples/PhoneTranslator/PhoneTranslator/ViewModels/MainViewModel.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Data;
using System.Windows.Documents;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Imaging;
using System.Windows.Shapes;
using System.Collections.ObjectModel;


namespace PhoneTranslator
{
    public class MainViewModel : INotifyPropertyChanged
    {
        public MainViewModel()
        {
            // Insert code required on object creation below this point
            Items = new ObservableCollection<ItemViewModel>() {
                       new ItemViewModel{Key="la", Value="Maecenas praesent accumsan bibendum"},
                       new ItemViewModel{Key="es", Value="Pharetra placerat pulvinar sagittis senectus sociosqu suscipit torquent ultrices vehicula volutpat maecenas praesent"},
                       new ItemViewModel{Key="kl",Value="Habitant inceptos interdum lobortis"},
                       new ItemViewModel{Key="c#", Value="foo =&gt; foo.bar"},
                       new ItemViewModel{Key="vb", Value="Function(foo) foo.bar"},
                       new ItemViewModel{Key="rb", Value="Senectus sociosqu suscipit torquent ultrices vehicula volutpat maecenas praesent accumsan bibendum dictumst eleifend"}};

        }

        public ObservableCollection<ItemViewModel> Items { get; private set; }

        private string sampleProperty = "Sample Runtime Property Value";
        /// <summary>
        /// Sample ViewModel property; this property is used in the view to display its value using a Binding
        /// </summary>
        /// <returns></returns>
        public string SampleProperty
        {
            get
            {
                return sampleProperty;
            }
            set
            {
                if (value != sampleProperty)
                {
                    sampleProperty = value;
                    NotifyPropertyChanged("SampleProperty");
                }
            }
        }

        /// <summary> Sample ViewModel method; this method is invoked by a Behavior that is associated with it in the View</summary>
        public void SampleMethod()
        {
            SampleProperty = "SampleMethod invoked.";
        }

        public event PropertyChangedEventHandler PropertyChanged;
        private void NotifyPropertyChanged(String propertyName)
        {
            PropertyChangedEventHandler handler = PropertyChanged;
            if (null != handler)
            {
                handler(this, new PropertyChangedEventArgs(propertyName));
            }
        }
    }
}

File: WP7RxSamplesVB/BingTranslatorRxVb/BingTranslatorRxVb/ViewModels/MainViewModel.vb
Imports System.ComponentModel
Imports System.Collections.ObjectModel

Public Class MainViewModel
    Implements INotifyPropertyChanged

    Public Property Items As ObservableCollection(Of ItemViewModel)

    Public Sub New()
        Items = New ObservableCollection(Of ItemViewModel) From {
                    New ItemViewModel With {.Key = "la", .Value = "Maecenas praesent accumsan bibendum"},
                    New ItemViewModel With {.Key = "es", .Value = "Pharetra placerat pulvinar sagittis senectus sociosqu suscipit torquent ultrices vehicula volutpat maecenas praesent"},
                    New ItemViewModel With {.Key = "kl", .Value = "Habitant inceptos interdum lobortis"},
                    New ItemViewModel With {.Key = "c#", .Value = "foo =&gt; foo.bar"},
                    New ItemViewModel With {.Key = "vb", .Value = "Function(foo) foo.bar"},
                    New ItemViewModel With {.Key = "rb", .Value = "Senectus sociosqu suscipit torquent ultrices vehicula volutpat maecenas praesent accumsan bibendum dictumst eleifend"}}
    End Sub


    Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
    Private Sub NotifyPropertyChanged(ByVal propertyName As String)
        Dim handler As PropertyChangedEventHandler = PropertyChangedEvent
        If handler IsNot Nothing Then
            handler(Me, New PropertyChangedEventArgs(propertyName))
        End If
    End Sub
End Class